x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/document_creation_logic.test.ts

```diff
@@ -163,5 +195,393 @@ describe('DocumentCreationLogic', () => {
+            documents: mockJson,
+          });
+          expect(DocumentCreationLogic.actions.setErrors).not.toHaveBeenCalled();
+        });
+
+        it('should accept a single JSON obj', () => {
+          const mockJson = { foo: 'bar' };
+          DocumentCreationLogic.actions.setTextInput('{"foo":"bar"}');
+          DocumentCreationLogic.actions.onSubmitJson();
+
+          expect(DocumentCreationLogic.actions.uploadDocuments).toHaveBeenCalledWith({
+            documents: [mockJson],
+          });
+          expect(DocumentCreationLogic.actions.setErrors).not.toHaveBeenCalled();
+        });
+      });
+    });
+
+    describe('uploadDocuments', () => {
+      describe('valid uploads', () => {
+        const mockValidDocuments = [{ foo: 'bar', bar: 'baz', qux: 'quux' }];
+        const mockValidResponse = {
+          errors: [],
+          validDocuments: { total: 3, examples: mockValidDocuments },
+          invalidDocuments: { total: 0, examples: [] },
+          newSchemaFields: ['foo', 'bar', 'qux'],
+        };
+
+        beforeAll(() => {
+          mount();
+          jest.spyOn(DocumentCreationLogic.actions, 'setSummary');
+          jest.spyOn(DocumentCreationLogic.actions, 'setCreationStep');
+        });
+
+        it('should set and show summary from the returned response', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(
+            Promise.resolve(mockValidResponse)
+          );
+
+          await DocumentCreationLogic.actions.uploadDocuments({ documents: mockValidDocuments });
+          await promise;
+
+          expect(DocumentCreationLogic.actions.setSummary).toHaveBeenCalledWith(mockValidResponse);
+          expect(DocumentCreationLogic.actions.setCreationStep).toHaveBeenCalledWith(
+            DocumentCreationStep.ShowSummary
+          );
+        });
+      });
+
+      describe('invalid uploads', () => {
+        beforeAll(() => {
+          mount();
+          jest.spyOn(DocumentCreationLogic.actions, 'setErrors');
+        });
+
+        it('handles API errors', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(
+            Promise.reject({
+              body: {
+                statusCode: 400,
+                error: 'Bad Request',
+                message: 'Invalid request payload JSON format',
+              },
+            })
+          );
+
+          await DocumentCreationLogic.actions.uploadDocuments({ documents: [{}] });
+          await promise;
+
+          expect(DocumentCreationLogic.actions.setErrors).toHaveBeenCalledWith(
+            '[400 Bad Request] Invalid request payload JSON format'
+          );
+        });
+
+        it('handles client-side errors', async () => {
+          const { http } = HttpLogic.values;
+          const promise = (http.post as jest.Mock).mockReturnValueOnce(new Error());
```

Comment:
I don't under stand this. How does `promise` get assign a promise value here?. You're return value that you are returning is note a promise 🤔

Reply:
I'm not actually sure.... but for some reason the test fails / doesn't work without it :dead_inside: I think it's because of the `await Promise.all(promises)` on line 187.